autoload/go/cmd.vim

```diff
@@ -124,6 +128,37 @@ function! go#cmd#Test(compile, ...)
     endif
 endfunction
 
+function! go#cmd#TestFunc(...)
+    " search flags legend (used only)
+    " 'b' search backward instead of forward
+    " 'c' accept a match at the cursor position
+    " 'n' do Not move the cursor
+    " 'W' don't wrap around the end of the file
+    "
+    " for the full list
+    " :help search
+    let test = search("func Test", "bcnW")
```

Comment:
Had to use `bcnW`, `s` + `n` is not allowed, but this does keep the cursor in place while still allowing to run the test if immediately on the test line, eg. `func TestSomething...`